### DISCUSSION TITLE
Regional snake latency reported wrong on profile page

### DISCUSSION THREAD

**Discussion by emilv** [BattlesnakeOfficial/feedback#144] [Bug Reports]
I have [a snake](https://play.battlesnake.com/u/lajm/crazy-eye-killer/) living in Gothenburg and set the region to GCP EU-WEST4. During gameplay the latency is steady at about 40 ms but the snake profile reports a "very high latency" of over 1000 ms!

![image](https://user-images.githubusercontent.com/1684914/143566316-b5eb53bd-18de-479e-b22b-e809a06ab284.png)
![image](https://user-images.githubusercontent.com/1684914/143566429-6419c462-c482-49ae-8d3b-46cac0628649.png)

I have tried refreshing the snake multiple times but keeps getting the same result.

The snake is a Nodejs application that is always running on my server (not an "app engine") so the problem is not caused by warmup on my side.

**Comment by robbles** [comment]
Thanks for the bug report @emilv! We'll look into what's going on here.

**Comment by robbles** [comment]
@emilv It looks to me like your Battlesnake is configured without a trailing forward slash, but the server redirects with HTTP 301 to add a slash when one is missing. When I hit the URL that's currently configured, I get this:

```
HTTP/1.1 301 Moved Permanently
Connection: keep-alive
Content-Length: 185
Content-Type: text/html
Date: Mon, 29 Nov 2021 18:40:57 GMT
Location: http://SNAKE_URL_REDACTED/
Server: nginx/1.14.2

<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.14.2</center>
</body>
</html>
```
(I've redacted the URL in the Location header above since this post is public, but it's the same URL, just with a trailing forward slash)

It looks like our code automatically follows this redirect, but it adds significant latency to the whole transaction. I tested it out with the trailing slash added to the URL, and I'm now getting a consistent 30-35 ms instead of 300-500 ms.

I haven't yet looked into our logs, but my guess is that the redirect is handled earlier in the flow (like in the API server instead of the proxy server) and several long-distance round trip times happen as a result.

**Comment by emilv** [comment]
@robbles That solved it :-)
